Flaky test: handles large transaction arrays with O(n) performance uses hard 10ms threshold

### Description

The test handles large transaction arrays with O(n) performance in packages/relay/tests/lib/ethGetBlockBy.spec.ts asserts a fixed < 10ms runtime for 20k items. 

This is highly sensitive to CI hardware, load, and Node/V8 variance, causing intermittent failures. The test is also labeled as “O(n)” but uses a single timing threshold, which doesn’t validate complexity.


Suggested fix: replace the hard threshold with a ratio‑based timing check (with warm‑up and median), or move performance assertions to a separate benchmark suite.

Test was introduced as part of: #4819

### Steps to reproduce

1. Run the relay test suite in a slower or loaded environment (e.g., CI, throttled VM).
2. Observe intermittent failure in:
    - packages/relay/tests/lib/ethGetBlockBy.spec.ts
    - test name: handles large transaction arrays with O(n) performance
3. Failure occurs when timing exceeds 10ms for 20k logs.

### Additional context

```
   1) eth_getBlockBy
       populateSyntheticTransactions w showDetails=true
         handles large transaction arrays with O(n) performance:
      AssertionError: expected 12.19496000000072 to be below 10
      + expected - actual
      -12.19496000000072
      +10
      
      at Context.<anonymous> (tests/lib/ethGetBlockBy.spec.ts:324:30)
      at Generator.next (<anonymous>)
      at /home/runner/_work/hiero-json-rpc-relay/hiero-json-rpc-relay/packages/relay/tests/lib/ethGetBlockBy.spec.ts:9:71
      at new Promise (<anonymous>)
      at __awaiter (tests/lib/ethGetBlockBy.spec.ts:5:12)
      at Context.<anonymous> (tests/lib/ethGetBlockBy.spec.ts:275:24)
      at processImmediate (node:internal/timers:485:21)
```

### Hedera network

_No response_

### Version

commit: 3428d7365487c8c9f3d84eb2a2ed3d0de41f3640

### Operating system

macOS

--- a/packages/relay/tests/lib/ethGetBlockBy.spec.ts
+++ b/packages/relay/tests/lib/ethGetBlockBy.spec.ts
@@ -306,22 +306,50 @@ describe('eth_getBlockBy', async function () {
     });
 
     it('handles large transaction arrays with O(n) performance', async function () {
+      const makeLogs = (count: number) =>
+        Array.from({ length: count }, (_, i) =>
+          mirrorLogToModelLog({
+            ...defaultLogs1[0],
+            transaction_hash: `0x${i.toString(16).padStart(64, '0')}`,
+            index: i,
+          }),
+        );
+
+      const runOnce = (logs: ReturnType<typeof makeLogs>) =>
+        __test__.__private.populateSyntheticTransactions(false, logs, [], '0x12a');
+
+      const median = (values: number[]) => {
+        const sorted = [...values].sort((a, b) => a - b);
+        const mid = Math.floor(sorted.length / 2);
+        return sorted.length % 2 === 0 ? (sorted[mid - 1] + sorted[mid]) / 2 : sorted[mid];
+      };
+
+      // Generate 10000 logs with unique hashes
+      const logsN = makeLogs(10000);
       // Generate 20000 logs with unique hashes
-      const largeLogs = Array.from({ length: 20000 }, (_, i) =>
-        mirrorLogToModelLog({
-          ...defaultLogs1[0],
-          transaction_hash: `0x${i.toString(16).padStart(64, '0')}`,
-          index: i,
-        }),
-      );
+      const logs2N = makeLogs(20000);
 
-      const start = performance.now();
-      const result = __test__.__private.populateSyntheticTransactions(false, largeLogs, [], '0x12a');
-      const duration = performance.now() - start;
+      // Warm-up
+      runOnce(logsN);
 
-      expect(result.length).to.equal(20000);
-      // With O(n) complexity, 20000 items should complete well under 10ms
-      expect(duration).to.be.lessThan(10);
+      const measureRuns = (logs: ReturnType<typeof makeLogs>, expectedLen: number, runs = 5) => {
+        const times: number[] = [];
+        for (let i = 0; i < runs; i++) {
+          const t0 = performance.now();
+          const result = runOnce(logs);
+          times.push(performance.now() - t0);
+          expect(result.length).to.equal(expectedLen);
+        }
+        return times;
+      };
+
+      const timesN = measureRuns(logsN, 10000);
+      const times2N = measureRuns(logs2N, 20000);
+
+      const ratio = median(times2N) / median(timesN);
+
+      // Loose bound to avoid flakiness but still catch superlinear behavior.
+      expect(ratio).to.be.lessThan(2.5);
     });
   });
 });

